src/main/java/com/autotune/analyzer/deployment/AutotuneDeployment.java

```diff
@@ -528,12 +528,27 @@ private static AutotuneConfig getAutotuneConfig(String autotuneConfigResource, K
 					}
 				}
 
-				String tunableName = tunableJson.optString(AutotuneConfigConstants.NAME);
-				String tunableValueType = tunableJson.optString(AutotuneConfigConstants.VALUE_TYPE);
-				String upperBound = tunableJson.optString(AutotuneConfigConstants.UPPER_BOUND);
-				String lowerBound = tunableJson.optString(AutotuneConfigConstants.LOWER_BOUND);
-				// Read in step from the tunable, set it to '1' if not specified.
-				double step = tunableJson.optDouble(AutotuneConfigConstants.STEP, 1);
+				String tunableName = tunableJson.optString(AnalyzerConstants.AutotuneConfigConstants.NAME);
+				String tunableValueType = tunableJson.optString(AnalyzerConstants.AutotuneConfigConstants.VALUE_TYPE);
+
+				String upperBound = "";
+				String lowerBound = "";
+				double step = 1;
+				List<String> choices = new ArrayList<>();
+
+				// check the tunable type, if it's categorical then we need to add the choices
+				if (tunableValueType.equalsIgnoreCase("categorical")) {
```

Comment:
@khansaad I would suggest you have a single check for categorical and have all of the code under that instead of having multiple checks for the same as below

Reply:
Updated